### PR TITLE
Fix #2299: Introduce append text fix

### LINKED ISSUE #2299: Some Espresso tests that require typing text are hanging
Some tests (e.g. StateFragmentTest) that type in text are hanging when run on Espresso. It seems that this might have been caused by the recent migration to EditTextInputAction since it removed the typeText() action we were previously using for these tests.

### PR DESCRIPTION
**Pull request by BenHenning**
Fix #2299.

This PR does the following:
- Updates EditTextInputAction to use ``typeText()`` when run on Espresso and ``appendText`` for Robolectric (see #2299 for context)
- Updates all references to ``typeText()`` to use EditTextInputAction

I've only verified this as working for StateFragmentTest.

**Before changes**
StateFragmentTest (on Espresso):

image
(note that the test actually takes longer than this, but it spent 5 minutes on a single test since it was hung)

StateFragmentTest (on Robolectric via Bazel):

image

**After changes**

StateFragmentTest (on Espresso):
image
(note that these failures are unrelated--it seems StateFragmentTest is trying to run ImageRegionSelection tests even though those aren't compatible with Espresso when run on a real device)

StateFragmentTest (on Robolectric via Bazel):
image

### CODE REVIEW

**Comment by BenHenning** [comment]
@anandwana001 if you can, could you please run all Espresso tests on this branch locally to make sure everything is now working per your expectations?

**Comment by anandwana001** [comment]
> @anandwana001 if you can, could you please run all Espresso tests on this branch locally to make sure everything is now working per your expectations?

Sure.

**Comment by anandwana001** [comment]
| Activity | Screenshot |
| --------|------------|
| `AddProfileActivityTest` | Screenshot 2020-12-23 at 13 45 58 |
| `AdminAuthActivityTest` | Screenshot 2020-12-23 at 14 13 09 | 
| `AdminPinActivityTest`| Screenshot 2020-12-23 at 14 17 33 |
| `PinPasswordActivityTest` | Screenshot 2020-12-23 at 14 22 55 |
| `ProfileResetPinActivityTest` | Screenshot 2020-12-23 at 14 25 50 |
| `ProfileRenameActivityTest` | Screenshot 2020-12-23 at 14 37 31 |
| `InputInteractionViewTestActivityTest` | Screenshot 2020-12-23 at 14 30 43 | 
| `StateFragmentLocalTest` | Screenshot 2020-12-23 at 14 35 04 | 
| `QuestionPlayerActivityLocalTest` | Screenshot 2020-12-23 at 14 39 10 |

**Review by anandwana001 — APPROVED** [review]
LGTM

**Comment by BenHenning** [comment]
Thanks @anandwana001! Will wait for @prayutsu until EOD tomorrow my time, otherwise will merge this.

**Review by prayutsu — APPROVED** [review]
LGTM! Thanks @BenHenning

**Comment by BenHenning** [comment]
Thanks @anandwana001 & @prayutsu!

**Comment by BenHenning** [comment]
@rt4914 FYI will need your code owner approval now.

**Comment by BenHenning** [comment]
@rt4914 / @anandwana001 please merge this once it's fully approved (since I'll be out).

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
